src/Symfony/Component/Security/Http/Authenticator/Passport/Badge/UserBadge.php

```diff
@@ -52,6 +52,10 @@ public function __construct(
         ?callable $userLoader = null,
         private ?array $attributes = null,
     ) {
+        if ('' === $userIdentifier) {
+            trigger_deprecation('symfony/security-http', '7.2', 'Using an empty string as user identifier is deprecated and will not be allowed in 8.0.');
```

Comment:
please add the line throwing in 8.0 in a comment

Reply:
@nicolas-grekas Sorry, I don't understand what changes you are requiring here.

Reply:
```suggestion
            trigger_deprecation('symfony/security-http', '7.2', 'Using an empty string as user identifier is deprecated and will not be allowed in 8.0. It will throw an exception in Symfony 8.0.');
```

Reply:
@OskarStark thanks.

Reply:
I mean that in 8.0, we will have a line that will throw instead of the deprecation. We should add this line right away, commented.

Reply:
Goal is to allow not forgetting about what needs to be done and just replace the deprecation by that line once the 8.0 branch exists (i.e. in a long time)

Reply:
@nicolas-grekas Done!